permissions/src/iosMain/kotlin/PermissionRefreshScheduler.kt

```diff
@@ -36,26 +38,26 @@ class PermissionRefreshScheduler<P : Permission>(
     coroutineScope: CoroutineScope = permissionManager
 ) : CoroutineScope by coroutineScope {
 
-    private sealed class TimerJobState(internal val coroutineScope: CoroutineScope) {
-        class TimerNotRunning(coroutineScope: CoroutineScope) : TimerJobState(coroutineScope) {
-            fun startTimer(interval: Long, block: suspend () -> Unit) = TimerRunning(interval, block, coroutineScope)
+    private sealed class TimerJobState() {
+        object TimerNotRunning : TimerJobState() {
+            fun startTimer(interval: Long, coroutineScope: CoroutineScope, block: suspend () -> Unit): TimerJobState = TimerRunning(interval, block, coroutineScope)
         }
 
-        class TimerRunning(val interval: Long, val block: suspend () -> Unit, coroutineScope: CoroutineScope) : TimerJobState(coroutineScope) {
+        class TimerRunning(val interval: Long, val block: suspend () -> Unit, coroutineScope: CoroutineScope) : TimerJobState() {
             private val timerLoop = coroutineScope.launch {
                 while (isActive) {
                     delay(interval)
                     block()
                 }
             }
 
-            fun stopTimer() = TimerNotRunning(coroutineScope).also { timerLoop.cancel() }
+            fun stopTimer() = TimerNotRunning.also { timerLoop.cancel() }
         }
     }
 
-    private var lastPermission: IOSPermissionsHelper.AuthorizationStatus? = null
-    var isWaiting: Boolean = false
-    private var timerState: TimerJobState = TimerJobState.TimerNotRunning(this)
+    private var lastPermission: AtomicReference<IOSPermissionsHelper.AuthorizationStatus?> = AtomicReference(null)
+    val isWaiting = AtomicBoolean(false)
+    private var timerState: AtomicReference<TimerJobState> = AtomicReference(TimerJobState.TimerNotRunning)
```

Comment:
So another thing is to be careful with `AtomicReference`, if you don't `null` them (so `set(null)` they can leak the value: https://kotlinlang.org/docs/mobile/concurrent-mutability.html#atomicreference

This is not a concern if it's a `boolean` or a simple `object` (such as `TimerNotRunning`), but it is if it's a running timer state. I assume this gets set back to NotRunning but it's not super easy to follow.

Reply:
Incidentally this can be `val` now, it seems.

Reply:
stopMonitoring will reset the TimerState back to TimerNotRunning, so it should be fine.